### PR TITLE
fix: some install(FILES..) assumed GLFW_BINARY_DIR

### PR DESCRIPTION
**Pull request by tamaskenez**
The `install(FILES ...)` command at line#587 mistakenly assumes the default path prefix is `CMAKE_CURRENT_BINARY_DIR` (where the source files are actually are) while it's `CMAKE_CURRENT_SOURCE_DIR`.

Before this commit the installation fails with this error message (CMake 3.2.1):

```
CMake Error at cmake_install.cmake:35 (file):
  file INSTALL cannot find
  "<GLFW_SOURCE_DIR>/lib/cmake/glfw3/glfw3Config.cmake".
```

### CODE REVIEW

**Comment by tamaskenez** [comment]
As a sidenote, this is the first time I see this separate-build-and-install-config-module feature in action. I'm curious, what is your use case which makes it worth the additional complexity? I mean:
- If you're actively working on GLFW (using it in another project), isn't it better to add it as a subdirectory instead of using build-tree config + user package registry?
- If you're just using GLFW without modifying it, it's a cleaner approach to use what is installed.

**Comment by shaxbee** [comment]
Hi Tomas,

The use case for build tree is project adding dependencies via
externalproject_add, then the targets can be refered to without installing.
Even for relatively simple projects like glfw install times are non-zero ;-)

Regards,
Zibi

File: CMakeLists.txt [hunk 1]
```diff
@@ -573,6 +573,10 @@ configure_file(src/glfw3.pc.in
 # The library is installed by src/CMakeLists.txt
 #--------------------------------------------------------------------
 if (GLFW_INSTALL)
+    if(IS_ABSOLUTE "${GLFW_INSTALL_CONFIG_DIR}")
```
Comment (shaxbee) [inline CMakeLists.txt]: This doesn't seem to be necessary - it is not an OPTION so shouldn't be overriden.

**Comment by tamaskenez** [comment]
Okay, I removed the check.

**Comment by tamaskenez** [comment]
shall I squash the two commits into one and issue a new pull-request?

**Comment by shaxbee** [comment]
No need. I'll cheery pick and squash them  tomorrow :-)

**Comment by shaxbee** [comment]
Cherry picked #4

**Pull request closed without merging**